lightning/src/ln/onion_utils.rs

```diff
@@ -878,7 +881,7 @@ impl HTLCFailReason {
 						network_update: None,
 						payment_failed_permanently: false,
 						short_channel_id: Some(path.hops[0].short_channel_id),
-						#[cfg(test)]
+						node_id: Some(path.hops[0].pubkey),
```

Comment:
We could, although then we don't have type safety that the error code won't contradict the bool. I didn't see any reason not to expose it since it's an internal struct, as long as the `vec` stays test-only.

Reply:
Actually, I think this does make more sense. It can be the case that the intro node errored but with the wrong error code so type safety isn't a concern. Will fix.